azavea/climate-change-lab
# Update Angular to latest point release

## Overview

Updating Angular (and associated packages) wasn't possible as part of #334 because the `climate-change-components` dependency needed to be updated first. Now that #336 is merged, this updates Angular to the latest point release.

### Notes

Upgrading Typescript to 2.6.2 (from 2.6.1) still caused `ERROR in AppModule is not an NgModule`. The suggested resolutions I've found all revolve around installing particular Typescript versions, or particular Angular versions; I haven't been able to find a root cause for the issue, which seems to have cropped up across many combinations of Typescript and Angular versions. It didn't seem worth the effort to dig into what's actually going on, so I think we should let that one sit, until we either need one of the [bug fixes in Typescript 2.6.2](https://github.com/Microsoft/TypeScript/issues?q=is%3Aissue+milestone%3A%22TypeScript+2.6.2%22+label%3A%22fixed%22+), or until we embark on a major-version upgrade.

## Testing Instructions

 * `yarn install`
 * `yarn run serve`
 * Make sure the frontend still functions properly

## Checklist

Finishes #295 (already closed)
